File: app/src/main/java/com/example/pcmall/service/CartService.java
package com.example.pcmall.service;

import com.example.pcmall.model.Cart;
import com.example.pcmall.model.response.ResponseResult;

import java.util.List;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.POST;

public interface CartService {
    @FormUrlEncoded
    @POST("searchCartList")
    Observable<ResponseResult<List<Cart>>> searchCartList(
            @Field("uid") String uid,
            @Field("currentPage") Integer currentPage,
            @Field("pageSize") Integer pageSize);

    @FormUrlEncoded
    @POST("getTotalCount")
    Observable<ResponseResult<Long>> getTotalCount(@Field("uid") String uid);

    @FormUrlEncoded
    @POST("addCartCount")
    public Observable<ResponseResult<String>> addCartCount(@Field("id") Integer id);

    @FormUrlEncoded
    @POST("subCartCount")
    public Observable<ResponseResult<String>> subCartCount(@Field("id") Integer id);

    @FormUrlEncoded
    @POST("selectCart")
    public Observable<ResponseResult<String>> selectCart(@Field("id") Integer id, @Field("isSelect") Integer isSelect);

    @FormUrlEncoded
    @POST("selectAllCart")
    public Observable<ResponseResult<String>> selectAllCart(@Field("uid") String uid, @Field("isSelect") Integer isSelect);

    @FormUrlEncoded
    @POST("addCart")
    public Observable<ResponseResult<String>> addCart(@Field("uid") String uid, @Field("gid") Integer gid);

    @FormUrlEncoded
    @POST("deleteCart")
    public Observable<ResponseResult<String>> deleteCart(@Field("id") Integer id);
}

File: core/network/src/main/java/com/example/pcmallcompose/core/network/service/CartService.kt
package com.example.pcmallcompose.core.network.service

import com.example.pcmallcompose.core.model.Cart
import com.example.pcmallcompose.core.model.response.ResponseResult
import io.reactivex.rxjava3.core.Observable
import retrofit2.http.Field
import retrofit2.http.FormUrlEncoded
import retrofit2.http.POST

interface CartService {
    @FormUrlEncoded
    @POST("searchAllCart")
    suspend fun searchAllCart(
        @Field("uid") uid: String,
        @Field("currentPage") currentPage: Int,
        @Field("pageSize") pageSize: Int
    ): ResponseResult<List<Cart>>

    @FormUrlEncoded
    @POST("searchSelectCart")
    fun searchSelectCart(@Field("uid") uid: String): Observable<ResponseResult<List<Cart>>>

    @FormUrlEncoded
    @POST("getTotalCount")
    suspend fun getTotalCount(@Field("uid") uid: String): ResponseResult<Long>

    @FormUrlEncoded
    @POST("getNormalSelectedCount")
    suspend fun getNormalSelectedCount(@Field("uid") uid: String): ResponseResult<Long>

    @FormUrlEncoded
    @POST("getNormalTotalCount")
    suspend fun getNormalTotalCount(@Field("uid") uid: String): ResponseResult<Long>

    @FormUrlEncoded
    @POST("addCartCount")
    suspend fun addCartCount(@Field("id") id: Int): ResponseResult<String>

    @FormUrlEncoded
    @POST("subCartCount")
    suspend fun subCartCount(@Field("id") id: Int): ResponseResult<String>

    @FormUrlEncoded
    @POST("selectCart")
    suspend fun selectCart(
        @Field("id") id: Int,
        @Field("isSelect") isSelect: Int
    ): ResponseResult<String>

    @FormUrlEncoded
    @POST("selectAllCart")
    suspend fun selectAllCart(
        @Field("uid") uid: String,
        @Field("isSelect") isSelect: Int
    ): ResponseResult<String>

    @FormUrlEncoded
    @POST("addCart")
    suspend fun addCart(@Field("gid") goodsId: Int, @Field("uid") uid: String): ResponseResult<String>

    @FormUrlEncoded
    @POST("deleteCart")
    suspend fun deleteCart(@Field("id") id: Int): ResponseResult<String>
}